.github/workflows/prettier.yml

```
@@ -0,0 +1,64 @@
+name: Format Code with Prettier
+
+on:
+  push:
+    branches:
+      - '*'
+      - '!develop'
+
+jobs:
+  format:
+    if: github.event_name == 'push'
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
+      pull-requests: write
+      issues: write
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+        with:
+          ref: ${{ github.event.pull_request.head.ref }}
+          repository: ${{ github.event.pull_request.head.repo.full_name }}
+          fetch-depth: 0
+          
+      - name: Set up Node.js
+        uses: actions/setup-node@v2
+        with:
+          node-version: '22'
+          cache: 'yarn'
+
+      - name: Cache node_modules
+        uses: actions/cache@v3
+        with:
+          path: node_modules
+          key: ${{ runner.os }}-node-modules-${{ hashFiles('**/yarn.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-node-modules-
+
+      - name: Install dependencies
+        run: yarn install
+
+      - name: Run Prettier
+        run: yarn prettier --write .
+
+      - name: Check for changes
+        id: check_changes
+        run: |
+          if git diff --quiet; then
+            echo "No code format changes detected"
+            echo "changes=false" >> $GITHUB_OUTPUT
+          else
+            echo "Code format changes detected"
+            echo "changes=true" >> $GITHUB_OUTPUT
+          fi
+
+      - name: Commit and push changes
+        if: steps.check_changes.outputs.changes == 'true'
```

Avoid infinite loop with something like this. It would work exactly like this if the actor to trigger the commit was github-actions.
```suggestion
        if: steps.check_changes.outputs.changes == 'true' && github.actor != 'github-actions[bot]'
```